src/ui/components/radioSelector/RadioSelector.tsx

```diff
@@ -0,0 +1,128 @@
+import React, { useRef } from 'react'
+import styled, { DefaultTheme } from 'styled-components/native'
+
+import { useHandleFocus } from 'libs/hooks/useHandleFocus'
+import { accessibleRadioProps } from 'shared/accessibilityProps/accessibleRadioProps'
+import { TouchableOpacity } from 'ui/components/TouchableOpacity'
+import { useArrowNavigationForRadioButton } from 'ui/hooks/useArrowNavigationForRadioButton'
+import { useSpaceBarAction } from 'ui/hooks/useSpaceBarAction.web'
+import { AccessibleIcon } from 'ui/svg/icons/types'
+import { ValidateOff } from 'ui/svg/icons/ValidateOff'
+import { RadioButtonSelected } from 'ui/svg/RadioButtonSelected'
+import { getSpacing, Typo } from 'ui/theme'
+
+export enum RadioSelectorType {
+  DEFAULT = 'default',
+  ACTIVE = 'active',
+  DISABLED = 'disabled',
+}
+
+interface ValidateOffIconProps extends AccessibleIcon {
+  type?: RadioSelectorType
+}
+
+interface RadioSelectorProps {
+  label: string
+  onPress: (label: string, type: RadioSelectorType) => void
+  description?: string
+  price?: string
+  type?: RadioSelectorType
+}
+
+export const RadioSelector = ({
+  label,
+  onPress,
+  description,
+  price,
+  type = RadioSelectorType.DEFAULT,
+}: RadioSelectorProps) => {
+  const containerRef = useRef(null)
+  const { onFocus, onBlur, isFocus } = useHandleFocus()
+
+  const handlePress = () => {
+    if (type === RadioSelectorType.DISABLED) {
+      return
+    }
+    onPress(label, RadioSelectorType.ACTIVE)
+  }
+
+  useArrowNavigationForRadioButton(containerRef)
+  useSpaceBarAction(isFocus ? handlePress : undefined)
+  return (
+    <RadioSelectorContainer
+      type={type}
+      {...accessibleRadioProps({
+        label,
+        checked: type === RadioSelectorType.ACTIVE,
+      })}
+      onPress={handlePress}
+      onFocus={onFocus}
+      onBlur={onBlur}>
+      <TextContainer>
+        <ButtonText type={type}>{label}</ButtonText>
+        {!!description && <Typo.CaptionNeutralInfo>{description}</Typo.CaptionNeutralInfo>}
+      </TextContainer>
+      <IconPriceContainer>
+        {!!price && <PriceText type={type}>{price}€</PriceText>}
+        {type === RadioSelectorType.ACTIVE ? (
+          <RadioButtonSelectedPrimary />
+        ) : (
+          <ValidateOffIcon type={type} />
+        )}
+      </IconPriceContainer>
+    </RadioSelectorContainer>
+  )
+}
+
+const RadioSelectorContainer = styled(TouchableOpacity)<{ type: RadioSelectorType }>(
+  ({ theme, type = RadioSelectorType.DEFAULT }) => {
+    let borderWidth = 1
+    let backgroundColor
+
+    if (type === RadioSelectorType.ACTIVE) {
+      borderWidth = 2
+    } else if (type === RadioSelectorType.DISABLED) {
+      borderWidth = 0
+      backgroundColor = theme.colors.greyLight
+    }
+
+    return {
+      flexDirection: 'row',
+      borderColor: theme.colors.greyDark,
+      borderRadius: theme.borderRadius.radius,
+      borderWidth,
+      padding: getSpacing(4),
+      backgroundColor,
+    }
+  }
+)
```

Comment:
Ouais alors ça je suis pas fan, c'est pas parlant.

```suggestion
const DisabledRadio = styled(TouchableOpacity)`...`
const ActiveRadio = styled(...)

const RadioSelectorContainer = (type) => {
  if (type === RadioSelectorType.DISABLED) return <DisabledRadio />
  if (type === RadioSelectorType.ACTIVE) return <ActiveRadio />
  return <DefaultRadio />
}
```

Là on mixe des trucs c'est un enfer.
Tu peux même avoir un `BaseRadioContainer` qui définit les props qui ne changeront pas (`flexDirection`, `borderColor`, ...) et tous tes autres variantes hériteront de ça.